### PR TITLE
sinkmanager(ticdc): consider table state when generate tasks

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/5928

### What is changed and how it works?
Consider the table state when generating tasks. If the table sink state is not replicating, then skip it.
#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/processor/sinkmanager/table_progress_heap.go [hunk 1]
```diff
@@ -45,7 +47,14 @@ func (p *progressHeap) Len() int {
 }
 
 func (p *progressHeap) Less(i, j int) bool {
-	return p.heap[i].nextLowerBoundPos.Compare(p.heap[j].nextLowerBoundPos) == -1
+	a := p.heap[i]
```
Comment (nongfushanquan) [inline cdc/processor/sinkmanager/table_progress_heap.go:50]: what will happen if two tables are both in removed state?

Comment (hi-rustin) [reply cdc/processor/sinkmanager/table_progress_heap.go:48]: Use ts to compare. Smaller higher priority.

File: cdc/processor/sinkmanager/table_sink_wrapper.go [hunk 2]
```diff
@@ -51,22 +53,31 @@ func newTableSinkWrapper(
...
+func (t *tableSinkWrapper) start() {
+	t.state.Store(tablepb.TableStateReplicating)
+}
+
 func (t *tableSinkWrapper) appendRowChangedEvents(events ...*model.RowChangedEvent) {
 	t.tableSink.AppendRowChangedEvents(events...)
 }
 
 func (t *tableSinkWrapper) updateReceivedSorterResolvedTs(ts model.Ts) {
+	if t.state.Load() == tablepb.TableStatePreparing && ts > t.startTs {
```
[the remark this answers is not in the document]
Comment (hi-rustin) [reply cdc/processor/sinkmanager/table_sink_wrapper.go:78]: When we add it. `AddTable`

**Review by hicqu — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**